tests/lib/Configuration/Provider/AlloyEditorTest.php

```diff
@@ -68,11 +69,165 @@ public function testGetConfiguration(): void
         self::assertEquals(
             [
                 'extraPlugins' => ['plugin1', 'plugin2'],
-                'extraButtons' => ['button1', 'button2'],
+                'toolbars' => [],
                 'classes' => ['class1', 'class2'],
                 'attributes' => ['attr1', 'attr2'],
             ],
             $provider->getConfiguration()
         );
     }
+
+    public function testToolbarConfiguration(): void
+    {
+        $configResolver = $this->createMock(ConfigResolverInterface::class);
+
+        $configResolver
+            ->method('hasParameter')
+            ->willReturn(true);
+
+        $configResolver
+            ->method('getParameter')
+            ->willReturnCallback(static function ($paramName) {
+                $map = [
+                    RichText::TOOLBARS_SA_SETTINGS_ID => [
+                        'test_toolbar' => [
+                            'buttons' => [
+                                'test_button' => [
+                                    'visible' => true,
+                                    'priority' => 0,
+                                ],
+                            ],
+                        ],
+                    ],
+                    RichText::CLASSES_SA_SETTINGS_ID => [],
+                    RichText::ATTRIBUTES_SA_SETTINGS_ID => [],
+                ];
+
+                return $map[$paramName] ?? null;
+            });
```

Comment:
This block is very similar in 3 places AFAICS. Could you extract it to common method with `buttons` part of config array as parameter, e.g. 
```php
private function buildTestToolbarReturnCallback(array $buttonsConfig): callable
```
?

Other approaches are of course welcomed as long as they avoid repetitions.